### PR TITLE
Nisha | Test | Provide support of master export in Giddh

### PR DESCRIPTION
**Pull request by nisha-walkover**
https://app.clickup.com/t/86cvqx99a

### CODE REVIEW

File: apps/web-giddh/src/app/shared/header/components/account-operations/account-operations.component.html [hunk 1]
```diff
@@ -209,3 +210,8 @@
 </div>
 <!-- ./ export ladger popup -->
 <!-- modals -->
+<div class="text-center master-export-btn overflow-hidden d-flex justify-content-center align-items-center" *ngIf="!((showEditGroup$ | async) || (activeGroupInProgress$ | async) ||    (showNewForm$ | async))">
+    <button mat-stroked-button color="primary" (click)="exportMasterDialog()">Export</button>
+</div>
```
Comment (dilpreetsinghofficial) [inline apps/web-giddh/src/app/shared/header/components/account-operations/account-operations.component.html:216]: format the code

File: apps/web-giddh/src/app/shared/header/components/account-operations/account-operations.component.ts [hunk 2]
```diff
@@ -131,9 +133,11 @@ export class AccountOperationsComponent implements OnInit, AfterViewInit, OnDest
     public groupExportLedgerBodyRequest: ExportBodyRequest = new ExportBodyRequest();
     /** List of discounts */
     public discounts: any[] = [];
+    /** Holds group unique name */
+    @Input() public currentGrpUniqueName: string = '';
```
Comment (dilpreetsinghofficial) [inline apps/web-giddh/src/app/shared/header/components/account-operations/account-operations.component.ts]: correct proper name

File: apps/web-giddh/src/app/shared/header/components/account-operations/account-operations.component.ts [hunk 3]
```diff
@@ -728,4 +732,18 @@ export class AccountOperationsComponent implements OnInit, AfterViewInit, OnDest
             }
         });
     }
+
+    /**
+     * Export master file
+     */
+    public exportMasterDialog() {
```
Comment (dilpreetsinghofficial) [inline apps/web-giddh/src/app/shared/header/components/account-operations/account-operations.component.ts]: Add return type

File: apps/web-giddh/src/app/shared/header/components/export-master-dialog/export-master-dialog.component.html [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+      <p class="pb-3">
+        {{ commonLocaleData?.app_note }} :
+        <span [innerHTML]="commonLocaleData.app_export_note">
+          {{ localeData?.export_note }}
+        </span>
+      </p>
+      <master-export-option
+        (exportFormMaster) = "exportForm = $event"
+      ></master-export-option>
+      <button mat-stroked-button color="primary" (click)="exportMaster()">Export</button>
```
Comment (dilpreetsinghofficial) [inline apps/web-giddh/src/app/shared/header/components/export-master-dialog/export-master-dialog.component.html]: Add translation

File: apps/web-giddh/src/app/shared/header/components/export-master-dialog/export-master-dialog.component.ts [hunk 5]
```diff
@@ -0,0 +1,122 @@
...
+    @Inject(MAT_DIALOG_DATA) public inputData,
+    private ledgerService: LedgerService,
+    private toaster: ToasterService,
+    private router: Router,
+    private dialogRef: MatDialogRef<ManageGroupsAccountsComponent>,
+    private store: Store<AppState>,
+    private groupWithAccountsAction: GroupWithAccountsAction
+  ) { }
+
+  ngOnInit() {
```
Comment (dilpreetsinghofficial) [inline apps/web-giddh/src/app/shared/header/components/export-master-dialog/export-master-dialog.component.ts]: Correct hook name and add return type

File: apps/web-giddh/src/app/shared/header/components/export-master-dialog/export-master-dialog.component.ts [hunk 6]
```diff
@@ -0,0 +1,122 @@
...
+  }
+
+
+
+  public exportMaster(): void {
```
Comment (dilpreetsinghofficial) [inline apps/web-giddh/src/app/shared/header/components/export-master-dialog/export-master-dialog.component.ts:44]: Add documentation

File: apps/web-giddh/src/app/shared/header/components/export-master-dialog/export-master-dialog.component.ts [hunk 7]
```diff
@@ -0,0 +1,122 @@
...
+    let exportRequest: ExportBodyRequest = new ExportBodyRequest();
+    exportRequest.exportType = this.inputData?.exportType;
+    exportRequest.columnsToExport = [];
+    const formValue = this.exportForm;
```
Comment (dilpreetsinghofficial) [inline apps/web-giddh/src/app/shared/header/components/export-master-dialog/export-master-dialog.component.ts]:
```suggestion
    const formValue = this.exportFormValue;
```

Correct in overall where it is using also check in parent for formValue

File: apps/web-giddh/src/app/shared/header/components/export-master-dialog/export-master-dialog.component.ts [hunk 8]
```diff
@@ -0,0 +1,122 @@
...
+    if (formValue.openingBalance) {
+      exportRequest.columnsToExport?.push("Opening Balance")
```
Comment (dilpreetsinghofficial) [inline apps/web-giddh/src/app/shared/header/components/export-master-dialog/export-master-dialog.component.ts]: Add semicolon in overall file where it is missing

File: apps/web-giddh/src/app/shared/header/components/export-master-dialog/export-master-dialog.component.ts [hunk 9]
```diff
@@ -0,0 +1,122 @@
...
+    }
+    if (formValue.swiftCode) {
+      exportRequest.columnsToExport?.push("Swift Code")
+    }
+    this.isLoading = true;
+    this.ledgerService.exportData(exportRequest).pipe(takeUntil(this.destroyed$)).subscribe((response) => {
+      this.isLoading = false;
+      if (response?.status === "success") {
+        this.toaster.showSnackBar("success", response?.body);
+        this.dialogRef.close();
```
Comment (dilpreetsinghofficial) [inline apps/web-giddh/src/app/shared/header/components/export-master-dialog/export-master-dialog.component.ts]:
```suggestion
        this.dialogRef?.close();
```

File: apps/web-giddh/src/app/shared/header/components/export-master-dialog/export-master-dialog.component.ts [hunk 10]
```diff
@@ -0,0 +1,122 @@
...
+        this.router.navigate(['pages/downloads'])
```
Comment (dilpreetsinghofficial) [inline apps/web-giddh/src/app/shared/header/components/export-master-dialog/export-master-dialog.component.ts]: Add semicolon

File: apps/web-giddh/src/app/shared/header/components/group-export-ledger-modal/export-group-ledger.component.html [hunk 11]
```diff
@@ -19,113 +19,133 @@ <h3>{{ localeData?.export_group_ledger }}</h3>
...
+                                [(ngModel)]="exportRequest.showDescription"
+                            >
+                                <span class="pull-left">{{
+                                    exportRequest.showDescription ? commonLocaleData?.app_yes : commonLocaleData?.app_no
+                                }}</span>
+                            </mat-slide-toggle>
+                        </label>
+                    </div>
+                </ng-container>
+                <ng-container *ngIf="expotyType === 'account'">
```
Comment (dilpreetsinghofficial) [inline apps/web-giddh/src/app/shared/header/components/group-export-ledger-modal/export-group-ledger.component.html]: Correct name

File: apps/web-giddh/src/app/shared/header/components/group-export-ledger-modal/export-group-ledger.component.ts [hunk 12]
```diff
@@ -23,6 +29,8 @@ export class ExportGroupLedgerComponent implements OnInit {
     /* This will hold common JSON data */
     @Input() public commonLocaleData: any = {};
     @Output() public closeExportGroupLedgerModal: EventEmitter<any> = new EventEmitter();
+    /** Holds group unique name */
+    @Input() public currentGrpUniqueName: string = '';
```
Comment (dilpreetsinghofficial) [inline apps/web-giddh/src/app/shared/header/components/group-export-ledger-modal/export-group-ledger.component.ts]: Correct also this

File: apps/web-giddh/src/app/shared/header/components/group-export-ledger-modal/export-group-ledger.component.ts [hunk 13]
```diff
@@ -71,9 +79,23 @@ export class ExportGroupLedgerComponent implements OnInit {
     }
     /** To hold export request object */
     public fileType: string = 'CSV';
+    /** hold exportType value */
+    public expotyType: string = 'ledger';
```
Comment (dilpreetsinghofficial) [inline apps/web-giddh/src/app/shared/header/components/group-export-ledger-modal/export-group-ledger.component.ts]: Check name in overall

File: apps/web-giddh/src/app/shared/header/components/group-export-ledger-modal/export-group-ledger.component.ts [hunk 14]
```diff
@@ -71,9 +79,23 @@ export class ExportGroupLedgerComponent implements OnInit {
...
+    /** True if api call in progress */
+    public isLoading: boolean = false;
+    /** Form Group for export  form */
```
Comment (dilpreetsinghofficial) [inline apps/web-giddh/src/app/shared/header/components/group-export-ledger-modal/export-group-ledger.component.ts]: format code

File: apps/web-giddh/src/app/shared/header/components/group-export-ledger-modal/export-group-ledger.component.ts [hunk 15]
```diff
@@ -110,9 +132,89 @@ export class ExportGroupLedgerComponent implements OnInit {
...
+            this.exportRequest.to = this.toDate;
+            this.closeExportGroupLedgerModal.emit({ from: this.fromDate, to: this.toDate, type: this.emailTypeSelected, fileType: this.fileType, order: this.order, body: this.exportRequest });
+        } else {
+            let exportRequest: ExportBodyRequest = new ExportBodyRequest();
+            exportRequest.exportType = "MASTER_EXPORT";
+            exportRequest.fileType = "CSV";
+            exportRequest.groupUniqueNames = [this.currentGrpUniqueName];
+            exportRequest.columnsToExport = [];
+            const formValue = this.exportForm;
+            if (formValue.openingBalance) {
```
Comment (dilpreetsinghofficial) [inline apps/web-giddh/src/app/shared/header/components/group-export-ledger-modal/export-group-ledger.component.ts:145]: Already given changes in previous file

File: apps/web-giddh/src/app/shared/header/components/master-export-option/master-export-option.component.ts [hunk 16]
```diff
@@ -0,0 +1,63 @@
...
+  private destroyed$: ReplaySubject<boolean> = new ReplaySubject(1);
+  /* This will hold local JSON data */
+  public localeData: any = {};
+  /* This will hold common JSON data */
+  public commonLocaleData: any = {};
+  constructor(
+    private formBuilder: FormBuilder
+  ) { }
+
+  ngOnInit() {
```
Comment (dilpreetsinghofficial) [inline apps/web-giddh/src/app/shared/header/components/master-export-option/master-export-option.component.ts]: Already given changes

File: apps/web-giddh/src/app/shared/header/components/master-export-option/master-export-option.component.ts [hunk 17]
```diff
@@ -0,0 +1,63 @@
...
+      address: new FormControl(false),
+      pinCode: new FormControl(false),
+      taxNumber: new FormControl(false),
+      partyType: new FormControl(false),
+      bankName: new FormControl(false),
+      bankAccountNumber: new FormControl(false),
+      ifscCode: new FormControl(false),
+      beneficiaryName: new FormControl(false),
+      branchName: new FormControl(false),
+      swiftCode: new FormControl(false),
```
Comment (dilpreetsinghofficial) [inline apps/web-giddh/src/app/shared/header/components/master-export-option/master-export-option.component.ts]: remove comma

File: apps/web-giddh/src/app/shared/header/components/master-export-option/master-export-option.component.ts [hunk 18]
```diff
@@ -0,0 +1,63 @@
...
+    })
+  }
+
+  public ngOnDestroy(): void {
```
Comment (dilpreetsinghofficial) [inline apps/web-giddh/src/app/shared/header/components/master-export-option/master-export-option.component.ts:74]: Add this document

File: apps/web-giddh/src/app/shared/header/components/master/master.component.scss [hunk 19]
```diff
@@ -1,8 +1,8 @@
-.list-item:focus,
-.list-item:focus .list-item {
-    color: var(--color-white) !important;
-    background: var(--color-master-page-sidebar-active-list);
-}
+// .list-item:focus,
+// .list-item:focus .list-item {
+//     color: var(--color-white) !important;
```
Comment (dilpreetsinghofficial) [inline apps/web-giddh/src/app/shared/header/components/master/master.component.scss]: remove commented code

File: apps/web-giddh/src/app/shared/header/components/master/master.component.ts [hunk 20]
```diff
@@ -65,6 +65,8 @@ export class MasterComponent implements OnInit, OnChanges, OnDestroy {
     private useAccountBreadcrumb: boolean = false;
     /** True if account has unsaved changes */
     private hasUnsavedChanges: boolean = false;
+    /** Emits groupUniqueName path */
+    @Output() public currentGrpUniqueName = new EventEmitter();
```
Comment (dilpreetsinghofficial) [inline apps/web-giddh/src/app/shared/header/components/master/master.component.ts]: Already given changes

File: apps/web-giddh/src/assets/locale/accounts/en.json [hunk 21]
```diff
@@ -77,5 +77,21 @@
     "export_group_ledger": "Export Group Ledger",
     "bank_details_error_message": "All the bank fields are mandatory if you provide data for any of them",
     "item_no": "Item #[ITEM_NO]",
-    "duplicate_email":"Duplicate email are not allowed"
+    "duplicate_email":"Duplicate email are not allowed",
+
```
Comment (dilpreetsinghofficial) [inline apps/web-giddh/src/assets/locale/accounts/en.json]: remove space

**Review by dilpreetsinghofficial — CHANGES_REQUESTED** [review]

File: apps/web-giddh/src/app/shared/header/components/account-operations/account-operations.component.html (hunk 1 above)
Comment (nisha-walkover) [reply apps/web-giddh/src/app/shared/header/components/account-operations/account-operations.component.html:216]: done

File: apps/web-giddh/src/app/shared/header/components/account-operations/account-operations.component.ts (hunk 2 above)
Comment (nisha-walkover) [reply apps/web-giddh/src/app/shared/header/components/account-operations/account-operations.component.ts]: done

File: apps/web-giddh/src/app/shared/header/components/account-operations/account-operations.component.ts (hunk 3 above)
Comment (nisha-walkover) [reply apps/web-giddh/src/app/shared/header/components/account-operations/account-operations.component.ts]: done

File: apps/web-giddh/src/app/shared/header/components/export-master-dialog/export-master-dialog.component.html (hunk 4 above)
Comment (nisha-walkover) [reply apps/web-giddh/src/app/shared/header/components/export-master-dialog/export-master-dialog.component.html]: done

File: apps/web-giddh/src/app/shared/header/components/export-master-dialog/export-master-dialog.component.ts (hunk 5 above)
Comment (nisha-walkover) [reply apps/web-giddh/src/app/shared/header/components/export-master-dialog/export-master-dialog.component.ts]: confirm

File: apps/web-giddh/src/app/shared/header/components/export-master-dialog/export-master-dialog.component.ts (hunk 6 above)
Comment (nisha-walkover) [reply apps/web-giddh/src/app/shared/header/components/export-master-dialog/export-master-dialog.component.ts:44]: done

File: apps/web-giddh/src/app/shared/header/components/export-master-dialog/export-master-dialog.component.ts (hunk 7 above)
Comment (nisha-walkover) [reply apps/web-giddh/src/app/shared/header/components/export-master-dialog/export-master-dialog.component.ts]: Done

File: apps/web-giddh/src/app/shared/header/components/export-master-dialog/export-master-dialog.component.ts (hunk 8 above)
Comment (nisha-walkover) [reply apps/web-giddh/src/app/shared/header/components/export-master-dialog/export-master-dialog.component.ts]: done

File: apps/web-giddh/src/app/shared/header/components/export-master-dialog/export-master-dialog.component.ts (hunk 9 above)
Comment (nisha-walkover) [reply apps/web-giddh/src/app/shared/header/components/export-master-dialog/export-master-dialog.component.ts]: done

File: apps/web-giddh/src/app/shared/header/components/group-export-ledger-modal/export-group-ledger.component.html (hunk 11 above)
Comment (nisha-walkover) [reply apps/web-giddh/src/app/shared/header/components/group-export-ledger-modal/export-group-ledger.component.html]: done

File: apps/web-giddh/src/app/shared/header/components/group-export-ledger-modal/export-group-ledger.component.ts (hunk 12 above)
Comment (nisha-walkover) [reply apps/web-giddh/src/app/shared/header/components/group-export-ledger-modal/export-group-ledger.component.ts]: done

File: apps/web-giddh/src/app/shared/header/components/group-export-ledger-modal/export-group-ledger.component.ts (hunk 13 above)
Comment (nisha-walkover) [reply apps/web-giddh/src/app/shared/header/components/group-export-ledger-modal/export-group-ledger.component.ts]: done

File: apps/web-giddh/src/app/shared/header/components/group-export-ledger-modal/export-group-ledger.component.ts (hunk 14 above)
Comment (nisha-walkover) [reply apps/web-giddh/src/app/shared/header/components/group-export-ledger-modal/export-group-ledger.component.ts]: done

File: apps/web-giddh/src/app/shared/header/components/group-export-ledger-modal/export-group-ledger.component.ts (hunk 15 above)
Comment (nisha-walkover) [reply apps/web-giddh/src/app/shared/header/components/group-export-ledger-modal/export-group-ledger.component.ts:145]: done

File: apps/web-giddh/src/app/shared/header/components/master-export-option/master-export-option.component.ts (hunk 16 above)
Comment (nisha-walkover) [reply apps/web-giddh/src/app/shared/header/components/master-export-option/master-export-option.component.ts]: done

File: apps/web-giddh/src/app/shared/header/components/master-export-option/master-export-option.component.ts (hunk 17 above)
Comment (nisha-walkover) [reply apps/web-giddh/src/app/shared/header/components/master-export-option/master-export-option.component.ts]: done

File: apps/web-giddh/src/app/shared/header/components/master-export-option/master-export-option.component.ts (hunk 18 above)
Comment (nisha-walkover) [reply apps/web-giddh/src/app/shared/header/components/master-export-option/master-export-option.component.ts:74]: done

File: apps/web-giddh/src/app/shared/header/components/master/master.component.scss (hunk 19 above)
Comment (nisha-walkover) [reply apps/web-giddh/src/app/shared/header/components/master/master.component.scss]: done

File: apps/web-giddh/src/app/shared/header/components/master/master.component.ts (hunk 20 above)
Comment (nisha-walkover) [reply apps/web-giddh/src/app/shared/header/components/master/master.component.ts]: done

File: apps/web-giddh/src/assets/locale/accounts/en.json (hunk 21 above)
Comment (nisha-walkover) [reply apps/web-giddh/src/assets/locale/accounts/en.json]: done

File: apps/web-giddh/src/app/new-inventory/component/export-inventory-master/export-inventory-master.component.ts [hunk 22]
```diff
@@ -14,7 +14,7 @@ import { MAT_DIALOG_DATA } from '@angular/material/dialog';
 })
 export class ExportInventoryMasterComponent implements OnInit {
   /** Form Group for export  form */
-  public exportForm: FormGroup;
+  public exportFormValue: FormGroup;
```
Comment (dilpreetsinghofficial) [inline apps/web-giddh/src/app/new-inventory/component/export-inventory-master/export-inventory-master.component.ts]: revert to old exportForm

File: apps/web-giddh/src/app/shared/header/components/account-operations/account-operations.component.ts [hunk 23]
```diff
@@ -131,9 +133,11 @@ export class AccountOperationsComponent implements OnInit, AfterViewInit, OnDest
     public groupExportLedgerBodyRequest: ExportBodyRequest = new ExportBodyRequest();
     /** List of discounts */
     public discounts: any[] = [];
+    /** Holds group unique name */
```
Comment (dilpreetsinghofficial) [inline apps/web-giddh/src/app/shared/header/components/account-operations/account-operations.component.ts]:
```suggestion
    /** Hold active group unique name */
```

File: apps/web-giddh/src/app/shared/header/components/export-master-dialog/export-master-dialog.component.ts [hunk 24]
```diff
@@ -0,0 +1,127 @@
...
+import { AppState } from 'apps/web-giddh/src/app/store';
+import { GroupWithAccountsAction } from 'apps/web-giddh/src/app/actions/groupwithaccounts.actions';
+
+@Component({
+  selector: 'app-export-master-dialog',
+  templateUrl: './export-master-dialog.component.html',
+  styleUrls: ['./export-master-dialog.component.scss']
+})
+export class ExportMasterDialogComponent implements OnInit {
+  /** Form Group for export  form */
```
Comment (dilpreetsinghofficial) [inline apps/web-giddh/src/app/shared/header/components/export-master-dialog/export-master-dialog.component.ts:19]: Hold export form group value

File: apps/web-giddh/src/app/shared/header/components/export-master-dialog/export-master-dialog.component.ts [hunk 25]
```diff
@@ -0,0 +1,127 @@
...
+    private router: Router,
+    private dialogRef: MatDialogRef<ManageGroupsAccountsComponent>,
+    private store: Store<AppState>,
+    private groupWithAccountsAction: GroupWithAccountsAction
+  ) { }
+
+  /**
+   * ngOnInit
+   */
+  public ngOnInit(): void {
```
Comment (dilpreetsinghofficial) [inline apps/web-giddh/src/app/shared/header/components/export-master-dialog/export-master-dialog.component.ts]: Remove if this is not using

File: apps/web-giddh/src/app/shared/header/components/group-export-ledger-modal/export-group-ledger.component.ts [hunk 26]
```diff
@@ -71,9 +79,23 @@ export class ExportGroupLedgerComponent implements OnInit {
     }
     /** To hold export request object */
     public fileType: string = 'CSV';
+    /** hold exportType value */
```
Comment (dilpreetsinghofficial) [inline apps/web-giddh/src/app/shared/header/components/group-export-ledger-modal/export-group-ledger.component.ts]:
```suggestion
    /** Hold export type */
```

File: apps/web-giddh/src/app/shared/header/components/master-export-option/master-export-option.component.ts [hunk 27]
```diff
@@ -0,0 +1,73 @@
...
+import { ReplaySubject, takeUntil } from 'rxjs';
+
+@Component({
+  selector: 'master-export-option',
+  templateUrl: './master-export-option.component.html',
+  styleUrls: ['./master-export-option.component.scss']
+})
+export class MasterExportOptionComponent implements OnInit {
+  /** Form Group for export  form */
+  public exportFormValue: FormGroup;
```
Comment (dilpreetsinghofficial) [inline apps/web-giddh/src/app/shared/header/components/master-export-option/master-export-option.component.ts]: Revert to old

File: apps/web-giddh/src/app/shared/header/components/master-export-option/master-export-option.component.ts [hunk 28]
```diff
@@ -0,0 +1,73 @@
...
+  /** Emits the show discount event  */
```
Comment (dilpreetsinghofficial) [inline apps/web-giddh/src/app/shared/header/components/master-export-option/master-export-option.component.ts]: Correct documentation

File: apps/web-giddh/src/app/shared/header/components/master-export-option/master-export-option.component.ts [hunk 29]
```diff
@@ -0,0 +1,73 @@
...
+  private destroyed$: ReplaySubject<boolean> = new ReplaySubject(1);
+  /* This will hold local JSON data */
+  public localeData: any = {};
+  /* This will hold common JSON data */
+  public commonLocaleData: any = {};
+  constructor(
+    private formBuilder: FormBuilder
+  ) { }
+
+  public ngOnInit(): void {
```
Comment (dilpreetsinghofficial) [inline apps/web-giddh/src/app/shared/header/components/master-export-option/master-export-option.component.ts:30]: Add documentation

File: apps/web-giddh/src/app/shared/header/components/master/master.component.ts [hunk 30]
```diff
@@ -65,6 +65,8 @@ export class MasterComponent implements OnInit, OnChanges, OnDestroy {
     private useAccountBreadcrumb: boolean = false;
     /** True if account has unsaved changes */
     private hasUnsavedChanges: boolean = false;
+    /** Emits groupUniqueName path */
+    @Output() public activeGroupUniqueName = new EventEmitter();
```
Comment (dilpreetsinghofficial) [inline apps/web-giddh/src/app/shared/header/components/master/master.component.ts]: Correct even emitter

File: apps/web-giddh/src/app/shared/header/components/new-manage-groups-accounts/manage-groups-accounts.component.ts [hunk 31]
```diff
@@ -49,6 +49,8 @@ export class ManageGroupsAccountsComponent implements OnInit, OnDestroy, AfterVi
     private hasUnsavedChanges: boolean = false;
     /** True if confirmation is open on search groups/accounts keyup event */
     private isPageLeaveConfirmationOpen: boolean = false;
+    /** shows group unique name */
```
Comment (dilpreetsinghofficial) [inline apps/web-giddh/src/app/shared/header/components/new-manage-groups-accounts/manage-groups-accounts.component.ts]: Correct documentation

**Review by dilpreetsinghofficial — CHANGES_REQUESTED** [review]

File: apps/web-giddh/src/app/new-inventory/component/export-inventory-master/export-inventory-master.component.ts (hunk 22 above)
Comment (nisha-walkover) [reply apps/web-giddh/src/app/new-inventory/component/export-inventory-master/export-inventory-master.component.ts]: done

File: apps/web-giddh/src/app/shared/header/components/account-operations/account-operations.component.ts (hunk 23 above)
Comment (nisha-walkover) [reply apps/web-giddh/src/app/shared/header/components/account-operations/account-operations.component.ts]: done

File: apps/web-giddh/src/app/shared/header/components/export-master-dialog/export-master-dialog.component.ts (hunk 24 above)
Comment (nisha-walkover) [reply apps/web-giddh/src/app/shared/header/components/export-master-dialog/export-master-dialog.component.ts:19]: done

File: apps/web-giddh/src/app/shared/header/components/export-master-dialog/export-master-dialog.component.ts (hunk 25 above)
Comment (nisha-walkover) [reply apps/web-giddh/src/app/shared/header/components/export-master-dialog/export-master-dialog.component.ts]: done

File: apps/web-giddh/src/app/shared/header/components/group-export-ledger-modal/export-group-ledger.component.ts (hunk 26 above)
Comment (nisha-walkover) [reply apps/web-giddh/src/app/shared/header/components/group-export-ledger-modal/export-group-ledger.component.ts]: done

File: apps/web-giddh/src/app/shared/header/components/master-export-option/master-export-option.component.ts (hunk 27 above)
Comment (nisha-walkover) [reply apps/web-giddh/src/app/shared/header/components/master-export-option/master-export-option.component.ts]: done

File: apps/web-giddh/src/app/shared/header/components/master-export-option/master-export-option.component.ts (hunk 29 above)
Comment (nisha-walkover) [reply apps/web-giddh/src/app/shared/header/components/master-export-option/master-export-option.component.ts:30]: done

File: apps/web-giddh/src/app/shared/header/components/master/master.component.ts (hunk 30 above)
Comment (nisha-walkover) [reply apps/web-giddh/src/app/shared/header/components/master/master.component.ts]: done

File: apps/web-giddh/src/app/shared/header/components/new-manage-groups-accounts/manage-groups-accounts.component.ts (hunk 31 above)
Comment (nisha-walkover) [reply apps/web-giddh/src/app/shared/header/components/new-manage-groups-accounts/manage-groups-accounts.component.ts]: done

**Review by dilpreetsinghofficial — APPROVED** [review]

File: apps/web-giddh/src/app/shared/header/components/account-operations/account-operations.component.ts [hunk 32]
```diff
@@ -728,4 +734,17 @@ export class AccountOperationsComponent implements OnInit, AfterViewInit, OnDest
             }
         });
     }
+
+    /**
+     * Export master file
```
Comment (ravinderwalkover) [inline apps/web-giddh/src/app/shared/header/components/account-operations/account-operations.component.ts]: JS doc is not proper. Use DocumentThis extension in VScode and do proper documentation with correct message

File: apps/web-giddh/src/app/shared/header/components/group-export-ledger-modal/export-group-ledger.component.ts [hunk 33]
```diff
@@ -110,9 +132,88 @@ export class ExportGroupLedgerComponent implements OnInit {
...
+            }
+            if (formValue.swiftCode) {
+                exportRequest.columnsToExport?.push("Swift Code");
+            }
+            this.isLoading = true;
+            this.ledgerService.exportData(exportRequest).pipe(takeUntil(this.destroyed$)).subscribe((response) => {
+                this.isLoading = false;
+                if (response?.status === "success") {
+                    this.toaster.showSnackBar("success", response?.body);
+                    this.closeExportGroupLedgerModal.emit();
```
Comment (ravinderwalkover) [inline apps/web-giddh/src/app/shared/header/components/group-export-ledger-modal/export-group-ledger.component.ts]: If you are using modal, do emit and navigate from parent component.

**Review by ravinderwalkover — CHANGES_REQUESTED** [review]

File: apps/web-giddh/src/app/shared/header/components/account-operations/account-operations.component.ts (hunk 32 above)
Comment (nisha-walkover) [reply apps/web-giddh/src/app/shared/header/components/account-operations/account-operations.component.ts]: done

File: apps/web-giddh/src/app/shared/header/components/group-export-ledger-modal/export-group-ledger.component.ts (hunk 33 above)
Comment (nisha-walkover) [reply apps/web-giddh/src/app/shared/header/components/group-export-ledger-modal/export-group-ledger.component.ts]: done

**Review by ravinderwalkover — APPROVED** [review]

**Pull request merged**